common/models/types/index.d.ts

```diff
@@ -17,6 +17,37 @@
  */	
 declare type USVString = string;
 
+/**
+ * Used to facilitate edit-distance calculations by allowing the LMLayer to
+ * efficiently search the model's lexicon in a Trie-like manner.
+ */
+declare interface LexiconTraversal {
+  /**
+   * Provides an iterable pattern used to search for words with a prefix matching
+   * the current traversal state's prefix when a new character is appended.
+   * 
+   * For example, if the current traversal state corresponds to 'th', children() may return
+   * an iterator with states corresponding 'e' (for 'the', 'then', 'there'),
+   * 'a' (for 'than', 'that'), etc.
+   * 
+   * @param char      The character suffixed to the existing lookup-prefix's string after
+   *                  any 'keying' transformations are applied.
```

Comment:
This description is very hard for me to follow. Could you maybe rewrite it?

I think it's saying that, if the traversal is active for `th`, then `char` will be `"a"` or `"e"`. I'm not sure how to succinctly describe this...

Another thing: I know `USVString` implies it, but could you please note that `char` can be an astral code points may, with a JavasScript `.length` of 2?

Reply:
> I think it's saying that, if the traversal is active for th, then char will be "a" or "e". I'm not sure how to succinctly describe this...

Neither do I.  That's why the description is what it is, and why it's hard to follow.

> Another thing: I know USVString implies it

It does?  I've never really been clear on why we made that typealias.

> but could you please note that char can be an astral code points may, with a JavasScript .length of 2?

So, "code point", so may be composed of two "code units".